libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFilterConfigurationState.ts

```diff
@@ -0,0 +1,62 @@
+// (C) 2024 GoodData Corporation
+import { useMemo } from "react";
+import { invariant } from "ts-invariant";
+
+import { IDashboardDateFilter, isUriRef } from "@gooddata/sdk-model";
+
+import { IDashboardDependentDateFilter } from "../../../../../../model/index.js";
+
+/**
+ * @internal
+ */
+export function useDependentDateFilterConfigurationState(
+    neighborFilters: IDashboardDateFilter[],
+    commonDateFilter?: IDashboardDateFilter,
+) {
+    const neighborDateFiltersWithDimensions: IDashboardDependentDateFilter[] = useMemo(() => {
+        return neighborFilters.map((neighborFilter) => {
+            const id = neighborFilter.dateFilter.dataSet;
+
+            const neighborFilterLocalId = isUriRef(id) ? id.uri : id!.identifier;
```

Comment:
Will there ever be URI ref? Tiger runs on ID only. Possibly invariant for this one could be used as well.

Reply:
Added invariant for it.